swig/cpp/src/Struct.cpp

```diff
@@ -247,161 +269,159 @@ Val::Val(int64_t int64_val, sr_type_t type) {
     _val = val;
     _deleter = S_Deleter(new Deleter(val));
 }
-Val::Val(uint8_t uint8_val, sr_type_t type) {
+Val::Val(uint8_t uint8_val) {
     sr_val_t *val = nullptr;
     val = (sr_val_t*) calloc(1, sizeof(sr_val_t));
-    if (val == nullptr)
+    if (val == nullptr) {
         throw_exception(SR_ERR_NOMEM);
-    if (type == SR_UINT8_T) {
-	val->data.uint8_val = uint8_val;
-    } else {
-        free(val);
-        throw_exception(SR_ERR_INVAL_ARG);
     }
-
-    val->type = type;
+    val->data.uint8_val = uint8_val;
+    val->type = SR_UINT8_T;
     _val = val;
     _deleter = S_Deleter(new Deleter(val));
 }
-Val::Val(uint16_t uint16_val, sr_type_t type) {
+Val::Val(uint16_t uint16_val) {
     sr_val_t *val = nullptr;
     val = (sr_val_t*) calloc(1, sizeof(sr_val_t));
-    if (val == nullptr)
+    if (val == nullptr) {
         throw_exception(SR_ERR_NOMEM);
-    if (type == SR_UINT16_T) {
-	val->data.uint16_val = uint16_val;
-    } else {
-        free(val);
-        throw_exception(SR_ERR_INVAL_ARG);
     }
-
-    val->type = type;
+    val->data.uint16_val = uint16_val;
+    val->type = SR_UINT16_T;
     _val = val;
     _deleter = S_Deleter(new Deleter(val));
 }
-Val::Val(uint32_t uint32_val, sr_type_t type) {
+Val::Val(uint32_t uint32_val) {
     sr_val_t *val = nullptr;
     val = (sr_val_t*) calloc(1, sizeof(sr_val_t));
-    if (val == nullptr)
+    if (val == nullptr) {
         throw_exception(SR_ERR_NOMEM);
-    if (type == SR_UINT32_T) {
-	val->data.uint32_val = uint32_val;
-    } else {
-        free(val);
-        throw_exception(SR_ERR_INVAL_ARG);
     }
-
-    val->type = type;
+    val->data.uint32_val = uint32_val;
+    val->type = SR_UINT32_T;
     _val = val;
     _deleter = S_Deleter(new Deleter(val));
 }
 Val::Val(uint64_t uint64_val, sr_type_t type) {
+    if (type != SR_UINT64_T) {
+        throw_exception(SR_ERR_INVAL_ARG);
+    }
     sr_val_t *val = nullptr;
     val = (sr_val_t*) calloc(1, sizeof(sr_val_t));
-    if (val == nullptr)
+    if (val == nullptr) {
         throw_exception(SR_ERR_NOMEM);
-    if (type == SR_UINT64_T) {
-	val->data.uint64_val = uint64_val;
-    } else {
-        free(val);
-        throw_exception(SR_ERR_INVAL_ARG);
     }
-
-    val->type = type;
+    val->data.uint64_val = uint64_val;
+    val->type = SR_UINT64_T;
     _val = val;
     _deleter = S_Deleter(new Deleter(val));
 }
 void Val::set(const char *xpath, const char *value, sr_type_t type) {
```

Comment:
The set methods that still have the type specified are for the language bindings (lua, python).
I think the char*, bool, decimal64, and int64 methods are the minimum required for the language bindings. I was not sure about whether uint64 was needed as well, I'm not familiar with integer conversions in lua/python and whether a uint64 with MSB set would get lost so I left the type field on uint64 too.

Reply:
@mislavn should know.

Reply:
Lua and Python SWIG bindings can only see int64, so that's why in SWIG I ignore the rest of the functions because from their perspective they have the same function signature. Also that's why int64 contains logic for setting all other integers.

Reply:
OK I'll remove explicit type args for uint64